node/chainSimulator/process/processor.go

```diff
@@ -106,17 +108,97 @@ func (creator *blocksCreator) CreateNewBlock() error {
 		return err
 	}
 
+	notNil := !check.IfNil(currentHeader)
```

Comment:
can you check if you can extract some of these sections of code into smaller functions?
e.g. here you could create a createProofForHeader(currentHeader, validators)